packages/expo-updates/src/UseUpdatesUtils.ts

```diff
@@ -0,0 +1,119 @@
+import * as Updates from './Updates';
+import type { UpdateEvent } from './Updates.types';
+import type {
+  AvailableUpdateInfo,
+  CurrentlyRunningInfo,
+  Manifest,
+  UpdatesInfo,
+  UseUpdatesCallbacksType,
+} from './UseUpdates.types';
+
+// The currently running info, constructed from Updates constants
+export const currentlyRunning: CurrentlyRunningInfo = {
+  updateId: Updates.updateId,
+  channel: Updates.channel,
+  createdAt: Updates.createdAt,
+  isEmbeddedLaunch: Updates.isEmbeddedLaunch,
+  isEmergencyLaunch: Updates.isEmergencyLaunch,
+  manifest: Updates.manifest,
+  runtimeVersion: Updates.runtimeVersion,
+};
+
+/////// Internal functions ////////
+
+// Constructs the availableUpdate from the update manifest
+export const availableUpdateFromManifest = (manifest: Manifest | null | undefined) => {
+  return manifest
+    ? {
+        updateId: manifest?.id ?? null,
+        createdAt:
+          manifest && 'createdAt' in manifest && manifest.createdAt
+            ? new Date(manifest.createdAt)
+            : manifest && 'publishedTime' in manifest && manifest.publishedTime
+            ? new Date(manifest.publishedTime)
+            : null,
+        manifest,
+      }
+    : undefined;
+};
+
+// Constructs the UpdatesInfo from an event
+export const updatesInfoFromEvent = (
+  updatesInfo: UpdatesInfo | undefined,
```

Comment:
```suggestion
  updatesInfo: UpdatesInfo,
```

Reply:
And then we can remove `currentlyRunning` from the three objects below as mentioned in https://github.com/expo/expo/pull/21880#discussion_r1154995204

Reply:
Fixed

Reply:
Fixed